syntax = "proto3";
package service;

message FileNode {
    string uid = 1;
    int32 nodeType =2;
    string type = 3;
    string path = 4;
    string name = 5;
    string hash = 6;
    bool broken = 7;
    repeated FileNode derivedFrom = 8;
    repeated InfoNode additionalInfo = 9;
}

message InfoNode {
    string uid = 1;
    int32 nodeType = 2;
    string type = 3;
    repeated Analyzer analyzer = 4;

    message DataNode {
        int32 nodeType = 1;
        string type = 2;
        string data = 3;
    }
    repeated DataNode dataNodes = 5;
}

message Analyzer {
    string uid = 1;
    string name = 2;
    int32 nodeType = 3;
}

message PathSubstitution {
    string old = 1;
    string new = 2;
}

message PackageNode {
    string uid = 1;
    string name = 2;
    string session = 3;
    int32 nodeType = 4;
    repeated FileNode targets = 5;
    repeated InfoNode additionalInfo = 6;
}
